Tools/TestWebKitAPI/Tests/WTF/CompactVariant.cpp

```diff
@@ -83,23 +83,39 @@ TEST(WTF_CompactVariant, Pointers)
 {
     int testInt = 1;
     float testFloat = 2.0f;
+    void* testPtr = reinterpret_cast<void*>(0x2600'1234'5678'9abc);
 
-    CompactVariant<int*, float*> variant = &testInt;
+    CompactVariant<int*, float*, void*> variant = &testInt;
     EXPECT_TRUE(WTF::holdsAlternative<int*>(variant));
     EXPECT_FALSE(WTF::holdsAlternative<float*>(variant));
+    EXPECT_FALSE(WTF::holdsAlternative<void*>(variant));
 
     WTF::switchOn(variant,
         [&](int* const& value)   { EXPECT_EQ(*value, 1); },
-        [&](float* const& value) { FAIL(); }
+        [&](float* const& value) { FAIL(); },
+        [&](void* const& value) { FAIL(); }
     );
 
     variant = &testFloat;
     EXPECT_FALSE(WTF::holdsAlternative<int*>(variant));
     EXPECT_TRUE(WTF::holdsAlternative<float*>(variant));
+    EXPECT_FALSE(WTF::holdsAlternative<void*>(variant));
 
     WTF::switchOn(variant,
         [&](int* const& value)   { FAIL(); },
-        [&](float* const& value) { EXPECT_EQ(*value, 2.0f); }
+        [&](float* const& value) { EXPECT_EQ(*value, 2.0f); },
+        [&](void* const& value)   { FAIL(); }
+    );
+
+    variant = testPtr;
+    EXPECT_FALSE(WTF::holdsAlternative<int*>(variant));
+    EXPECT_FALSE(WTF::holdsAlternative<float*>(variant));
+    EXPECT_TRUE(WTF::holdsAlternative<void*>(variant));
+
+    WTF::switchOn(variant,
+        [&](int* const& value)   { FAIL(); },
+        [&](float* const& value) { FAIL(); },
+        [&](void* const value) { EXPECT_EQ(reinterpret_cast<uintptr_t>(value), 0x2600'1234'5678'9abcUL); }
```

Comment:
Pretty sure this will fail on 32-bit platforms.

Reply:
This test now only builds on 64-bit platforms.